DRAFT.md

```diff
@@ -1173,25 +1173,30 @@ outside of tests.
 
 A converting constructor could be added in a future version of the C++ standard.
 
-### Small object optimization for `polymorphic`
-
-`polymorphic` could be designed to include a small object optimization like that
-used in `std::string`. A small object optimization uses a buffer to potentially
-store the owned object and avoid allocating memory. This would make move
-construction more complicated as the owned object must be moved from one buffer
-to another, potentially invoking allocations if the owned object's move
-constructor allocates memory.
-
-As designed, `polymorphic<T>` does not require that `T` (or constructed classes
-of type `U` derived from `T`) are move constructible or move assignable for
-`polymorphic<T>` to be move constructible or move assignable.
-
-A polymorphic value type with a small buffer optimization that did not allocate
-a control block for the owned object would need to be a different type. It is
-not possible to add a small object optimization to `polymorphic` without making
-breaking changes. There may be a case for the addition of `small_polymorphic<T,
-N>` similar to `llvm::SmallVector<T, N>`, but we are not proposing its addition
-here.
+### Small Buffer Optimisation
+
+It is possible to implement `polymorphic` with a small buffer optimisation,
+similar to that used in `std::function`. This would allow `polymorphic` to store
+small objects without allocating memory. Like `std::function`, the size of the
+small buffer is left to be specified by the implementation. Allowing a small
+buffer optimization means that constructors and potentially allocating member
+functions of polymorphic cannot be marked `constexpr`.
```

Comment:
As discussed, we do not believe this to be an inherent problem with the language not allowing `constexpr` here but rather a result of the implementation using variant. 

The following paper added support for `constexpr` variants in C++23 but yet to be implemented by all compilers: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2231r1.html

Reply:
I've rewritten this section now, please re-review.